scripts/upgrade

```diff
@@ -98,27 +127,21 @@ ynh_add_nginx_config
 # Set permissions to app files
 chown -R root: $final_path
 
-#=================================================
-# SETUP SSOWAT
-#=================================================
-ynh_script_progression --message="Upgrading SSOwat configuration..."
-
-# Make app public if necessary
-if [ $is_public -eq 1 ]
-then
-	# unprotected_uris allows SSO credentials to be passed anyway.
-	ynh_app_setting_set --app=$app --key=skipped_uris --value="/"
-fi
-
 #=================================================
 # RELOAD NGINX
 #=================================================
-ynh_script_progression --message="Reloading nginx web server..."
+ynh_script_progression --message="Reloading nginx web server..." --weight=1
 
 ynh_systemd_action --service_name=nginx --action=reload
 
 #=================================================
 # END OF SCRIPT
 #=================================================
 
+if [ ! -z $show_warning ]; then
+	if [ "$show_warning" = "1" ]; then
+                ynh_print_warn --message="Due to the fact that Yunohost resets permissions when upgrading a package from the old permissions system to the new one, we had to define your instance of Hextris as public. If you want to make it private, you can do it via webmin."
+	fi
+fi
+
```

Comment:
Ok, it wasn't just your debug ! Sorry ;)
Same comment as before about `-z` and `= "1"`

Reply:
> But I have to admit that I don't understand why the app can't stay private...

https://github.com/YunoHost-Apps/hextris_ynh/pull/13#discussion_r400146128
If, when the installation, the app is installed as private, it will stay private. But if it was installed public, and after the upgrade to ynh 3.7 was made private with the new perm system, I've to make it private due to what kayou said before.

Also, I followed the example_ynh's permission branch. So I think you'll need to update this app to make changes you proposed propagate.

Reply:
Many apps use: https://github.com/YunoHost/example_ynh/blob/15bf6f665e108a6086122158ae7d042660b66626/scripts/install#L224 to finish their configuration.

That's why, in this helper: https://github.com/YunoHost/yunohost/blob/13b510b6249af25f2b0289df10a0b35baef90f7a/data/helpers.d/setting#L200 when the key "skipped_url" is added with the url "/", we add the `visitors` group to the permission, and remove it when we delete this key... It's not a perfect solution, we still can modify it if you have a better one :-)

Reply:
We can't go on with that current situation with this warning at the end etc, it's not suitable for all apps.
If I understand well, the problem here is that when you call `ynh_app_setting_delete --app=$app --key=skipped_uris` it removed the visitors group. So the app become private.

But, that's a migration part of the upgrade script.
The key `is_public` is removed during the migration. https://github.com/YunoHost-Apps/hextris_ynh/pull/13/files#diff-44cb16c778719320333118c04d509a7cR69
So we know if the migration has been done or not.

This line should be removed https://github.com/YunoHost-Apps/hextris_ynh/pull/13/files#diff-44cb16c778719320333118c04d509a7cR53
And the condition part should look like that:
```
if [ -n "$is_public" ]; then
        ynh_app_setting_delete --app=$app --key=skipped_uris
	if [ $is_public -eq 1 ]; then
		ynh_permission_update --permission "main" --add "visitors"
        fi
	ynh_app_setting_delete --app=$app --key=is_public
fi
```

That way, if the app was public, we add the permission for visitors.
In any case, we remove the skipped_uris uris once, during the first migration of the app.
And the permission is set back once as well.

The key `is_public` is removed after that migration, the code won't be used anymore.

Reply:
I made the warn because if :
- I install the app with is_public=true on yunohost < 3.7
- I upgrade yunohost to 3.7
- I change my mind regarding whether I want the app to be public and using the new permission system, I make the app private
- Then I install this upgrade

The app will be made public by the upgrade script while I explicitly said I want it to be private (step 3)

Reply:
And apparently, there's no helper to check the current status of permission. Only its existence...

Reply:
@maniackcrudelis your proposal sounds good, I can modify the example accordingly. Warning when we add the visitor group back is a good thing too I guess. (at the same place)

Reply:
But hard to tell what kind of warning when we don't know what was done before, if even something was done.

Also the point is to keep it clear and as simple as possible.

Reply:
```bash
# Cleaning legacy permissions
is_public=$(ynh_app_setting_get --app=$app --key=is_public)

if [ -n "$is_public" ]; then
	# Remove skipped_uris
	ynh_app_setting_delete --app=$app --key=skipped_uris
	# Remove unprotected_uris
	ynh_app_setting_delete --app=$app --key=unprotected_uris
	# Remove protected_uris
	ynh_app_setting_delete --app=$app --key=protected_uris

	# Removing skipped/unprotected_uris under certain conditions, remove the visitors group added during the migration process of 3.7
	# If the app was public, add visitors again to the main permission
	if [ $is_public -eq 1 ]; then
		ynh_print_warn --message="Due to the migration of the legacy to the new authorization system, we are forcing $app to be public. You can manage permissions more precisely on the YunoHost admin panel."
		ynh_permission_update --permission "main" --add "visitors"
	fi
	ynh_app_setting_delete --app=$app --key=is_public
fi
```

Reply:
Yeah there is no way to get the current permission status. We can add one if we feel the need in a minor release, like 3.7.1 (we are in 3.7.0.12 \o/)

Reply:
First I think only skipped_uris should be into the condition, as other does not have this behavior and can be used even if the app was private (situation with public part even if the app is private).
Then, my point is, you add a warning to says that is forced to be public. But the app was, supposedly, public already. We don't know if we have changed anything here. To add such a warning can be confusing.

Reply:
> First I think only skipped_uris should be into the condition, as other does not have this behavior and can be used even if the app was private (situation with public part even if the app is private).

Yeah sorry, I was talking for the `example_ynh` project

>  Then, my point is, you add a warning to says that is forced to be public. But the app was, supposedly, public already. We don't know if we have changed anything here. To add such a warning can be confusing.

An admin can change permission of an app, then don't understand why after an upgrade (could be in 6 months or some apps) the app is public again

Reply:
> Yeah sorry, I was talking for the example_ynh project

Me too.

Well, you got a point. Then I guess the only real good solution is to know the status of a permission. Anyway, I really do think that sooner or later we're going to have scripts that does need to handle some permissions.
So we're going to need such a helper.
Also, we can't just assume that a part as important as this one of an app can be handle only on core side...

Reply:
> Also, we can't just assume that a part as important as this one of an app can be handle only on core side...

Well for now apart from the regex part and the fact that we can't protect a permission I don't see anything else we can't do in the new permission system. (I'm working with Josue on these point).

Anyway we should continue our discussion here: https://github.com/YunoHost/example_ynh/pull/111 :-)